packages/sign_in_with_apple/lib/src/exceptions.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
+@immutable
+abstract class SignInWithAppleException implements Exception {
```

Comment:
@tp I also started out creating some proper exception types which should make the error handling more pleasant for the users and easier to discover which errors could occur.

I'm not sure how fine-grained these need to be though.

Reply:
Probably  a little too specific in practice, but also does no harm as everything is under the `SignInWithAppleException` umbrella.

I think the only thing that is relevant is to clearly separate "something is broken" (like no SiwA support due to now having 2FA set up) from "the user cancelled".

Reply:
I think we can further improve the docs here in a follow up PR